### PR TITLE
Android publisher

### PR DESCRIPTION
**Pull request by rflechner**
Hi,

I developed a Fake Android publisher module to help android developers to automatically publish their APKs

I didn't implement expansion files for the moment, so maximum APK size uploadable by the module is 1 GB.

There is 2 different commits: 
	- this first is for the module implementation.
	- the second is for documentation and tutorial.

Kind regards,

Romain.

### CODE REVIEW

File: src/app/FakeLib/AndroidPublisher.fs [hunk 1]
```diff
@@ -0,0 +1,363 @@
+﻿(*
+    This module will help android developers to publish automatically their APKs
+
+    Usefull links:
```
Comment (forki) [inline src/app/FakeLib/AndroidPublisher.fs]: small typo

File: src/app/FakeLib/AndroidPublisher.fs [hunk 2]
```diff
@@ -0,0 +1,363 @@
+﻿(*
+    This module will help android developers to publish automatically their APKs
```
Comment (forki) [inline src/app/FakeLib/AndroidPublisher.fs]: "this module helps"

**Comment by forki** [comment]
wow. very interesting

File: androidpublisher.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
+# Publish Android apk
+
+This module will help android developers to automatically publish their APKs
```
Comment (forki) [inline androidpublisher.md]: "This module helps"

File: androidpublisher.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+## Prerequisite
+
+Before using this module, you will need an android keystore for apk signing.
+
+Next, you will need a Google service account decribed here:
```
Comment (forki) [inline androidpublisher.md]: described

File: androidpublisher.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+	//Clean old apk
+	Target "Clean" (fun _ ->
+		CleanDir androidBuildDir
+		CleanDir androidProdDir
+	)
+
+	Target "android-package" (fun () ->
+		AndroidPackage(fun defaults ->
+						{ defaults with
```
Comment (forki) [inline androidpublisher.md]: please use 4-spaces for indent. this will be better readably on small devices.

File: androidpublisher.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+			{ defaults with 
+				KeystorePath = @"path to my file.keystore"
+				KeystorePassword = "my password"
+				KeystoreAlias = "my key alias"
+			})
+		|> fun file -> file.CopyTo(Path.Combine(androidProdDir, file.Name)) |> ignore
+
+	)
+
+	Target "publish" (fun _ ->
```
Comment (forki) [inline androidpublisher.md]: "Convention" is to use uppercase for target names

File: androidpublisher.md [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+        let packageName = "my Android package name"
+
+		// to publish an alpha version: 
+        PublishApk { AlphaSettings with Certificate = certificate; PackageName = packageName; AccountId = mail; Apk = apkPath; }
+
+		// to publish a beta version: 
+        // PublishApk { BetaSettings with Certificate = certificate; PackageName = packageName; AccountId = mail; Apk = apkPath; }
+
+		// to publish a production version: 
+        // PublishApk { ProductionSettings with Certificate = certificate; PackageName = packageName; AccountId = mail; Apk = apkPath; }
```
Comment (forki) [inline androidpublisher.md]: maybe you could refactor Certificate, PackageName, AccountId, ApkPath into a sub-record!?

File: src/app/FakeLib/AndroidPublisher.fs [hunk 8]
```diff
@@ -0,0 +1,363 @@
...
+
+    override x.GetWebRequest (address:Uri) =
+        let rq = base.GetWebRequest address
+        rq.Timeout <- int32 <| TimeSpan.FromHours(1.).TotalMilliseconds
+        rq
+
+    member x.CreateRequest (address:Uri) =
+        x.GetWebRequest (address)
+
+let private AndroidPublisherScope = "https://www.googleapis.com/auth/androidpublisher"
```
Comment (forki) [inline src/app/FakeLib/AndroidPublisher.fs]: are these URLs stable? Maybe we should make them public and mutable!?

**Comment by rflechner** [comment]
Hi,

I thank you for the code review.

I changed those lines as you suggested.

Do I create another pull request ?

**Comment by forki** [comment]
no you can just append to this one.

* Commit your changes to your local repo
* git push origin AndroidPublisher
* comment on this thread, so that I know.

**Comment by forki** [comment]
I still need to review the actual code, but I'm on vacation for a couple of days.
Will try to look into this on tuesday

**Comment by rflechner** [comment]
Ok.

I think i will refactor uploadApk function and implement more features in next pull request.

Have a great holiday.

**Comment by forki** [comment]
thanks. Please try it out.

**Pull request merged**